__tests__/main.test.ts

```diff
@@ -0,0 +1,379 @@
+        expect(PipelineHelper.getErrorAndWarningMessageFromBuildResult(
+            { serverError: { message: 'ServerErrorMessage'  } } as any
+        )).toMatchObject({
+            errorMessage: 'ServerErrorMessage',
+            warningMessage: '',
+        });
+    });
+});
+
+describe('Testing all functions of class UrlParser', () => {
+    test('UrlParser.GetProjectName() - return project name from project URL', () => {
+        expect(UrlParser.GetProjectName('https://dev.azure.com/organization/project-name ')).toBe('project-name');
+    });
+
+    test('UrlParser.GetProjectName() - throw error if null or empty', () => {
+        expect(() => UrlParser.GetProjectName(null)).toThrow('Project url is null or empty. Specify the valid project url and try again');
+        expect(() => UrlParser.GetProjectName('')).toThrow('Project url is null or empty. Specify the valid project url and try again');
+    });
+
+    test('UrlParser.GetProjectName() - throw error if invalid url', () => {
+        expect(() => UrlParser.GetProjectName('https://dev.azure.com/organization/project-name/')).toThrow(`Failed to parse project url: "${'https://dev.azure.com/organization/project-name/'}". Specify the valid project url (eg, https://dev.azure.com/organization/project-name or https://server.example.com:8080/tfs/DefaultCollection/project-name)) and try again.`);
+        expect(() => UrlParser.GetProjectName('https://dev.azure.com/organization//')).toThrow(`Failed to parse project url: "${'https://dev.azure.com/organization//'}". Specify the valid project url (eg, https://dev.azure.com/organization/project-name or https://server.example.com:8080/tfs/DefaultCollection/project-name)) and try again.`);
+    });
+
+    test('UrlParser.GetCollectionUrlBase() - return collections base URL', () => {
+        expect(UrlParser.GetCollectionUrlBase('https://dev.azure.com/organization/project-name ')).toBe('https://dev.azure.com/organization');
+    });
+
+    test('UrlParser.GetCollectionUrlBase() - throw error if null or empty', () => {
+        expect(() => UrlParser.GetCollectionUrlBase(null)).toThrow('Project url is null or empty. Specify the valid project url and try again');
+        expect(() => UrlParser.GetCollectionUrlBase('')).toThrow('Project url is null or empty. Specify the valid project url and try again');
+    });
+
+    test('UrlParser.GetCollectionUrlBase() - throw error if invalid url', () => {
+        expect(() =>  UrlParser.GetCollectionUrlBase('/')).toThrow(`Failed to parse project url: "${'/'}". Specify the valid project url (eg, https://dev.azure.com/organization/project-name or https://server.example.com:8080/tfs/DefaultCollection/project-name)) and try again.`);
+    });
+});
+
+describe('Testing all functions of class PipelineRunner', () => {
+    test('start() - run a build using given parameters', async () => {
```

Comment:
what is being tested here ?

Reply:
go through a regular run to show what functions are called and cover that part of code.